commands/release/release.go

```diff
@@ -91,6 +90,67 @@ var ReleaseCmd = &cobra.Command{
 
 var emptyDomain []string = nil
 
+func parseChangeLogFile(changelogFile []byte) (bool, error) {
+	tbdHeaderFound := false
+	isBreakingChange := false
+
+	versionHeaderBool := false
+	nonEmptyLineFoundBool := false
+	scanner := bufio.NewScanner(bytes.NewReader(changelogFile))
+
+	for scanner.Scan() {
+		// Check if TBD is the first non-empty line - this is for extra caution.
+		if !emptyLineRegex.Match(scanner.Bytes()) {
+			if !versionToBeReleasedPlaceholderHeaderRegex.Match(scanner.Bytes()) {
+				return false, stacktrace.NewError("TBD header is either missing or is not the first non empty line in changelog.md")
+			}
+			tbdHeaderFound = true
+			break
+		}
+	}
+
+	// No TBD header was found because the file is empty.
+	if !tbdHeaderFound {
+		return false, stacktrace.NewError("Empty changelog file, please check the filepath again.")
+	}
+
+	for scanner.Scan() {
+		if versionToBeReleasedPlaceholderHeaderRegex.Match(scanner.Bytes()) {
+			return false, stacktrace.NewError("Found more than %d TBD headers, there can only be #d TBD header in the changelog", expectedNumTBDHeaderLines)
+		}
+
+		// Scan file until next version header detected, searching for first not empty line along the way
+		if versionHeaderRegex.Match(scanner.Bytes()) {
+			versionHeaderBool = true
+			break
+		}
+
+		if !emptyLineRegex.Match(scanner.Bytes()) {
+			nonEmptyLineFoundBool = true
+		}
+
+		// there exist breaking change header between TBD and last released version
+		if breakingChangesRegex.Match(scanner.Bytes()) {
+			isBreakingChange = true
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return false, stacktrace.Propagate(err, "An error occurred while scanning the bytes of the changelog file.")
+	}
+
+	if !versionHeaderBool {
```

Comment:
Thisis why we make the bools questions - because `!versionHeaderBool` is a bit ambiguous, but `!foundLastReleasedVersionHeader` is very clear